### PR TITLE
MNTOR-3233/add internal dashboard stuck removals

### PR DESCRIPTION
**Pull request by rhelmer**
# References:

Jira: MNTOR-3233

# Description

Add an admin dashboard to view and API for CSV export of slow removals. For any removals over 30 days, show:
- Scan Result ID
- Removal Status
- Created
- Last Updated
- Broker Name
- Broker Status

# Screenshot (if applicable)

image

# How to test

You can modify the HTTP parameters in the URL, the `days` and `perPage` are probably the most interesting, you can test pagination that way. Pagination uses the database but I used the `<Link>` component so the client seems pretty smart about not reloading the whole page.

I've been testing by modifying my local database, since the sandbox doesn't provide very interesting data:

```
blurts=# update onerep_scan_results set status = 'optout_in_progress'; update onerep_scan_results set updated_at = NOW() - INTERVAL '45 days'; update onerep_scan_results set created_at = NOW() - INTERVAL '45 days';
```

NOTE: loading the dashboard will re-sync the data from the sandbox to your local database.

Refreshing the data against the provider API can take a while, I've provided a link to do this, currently it just does a `GET` against the API with the appropriate parameters, so you need to hit the Back button to get back to the UI. If this is annoying enough we can make the UI handle this API call, but I wanted to keep the admin UI simple and not drag too much into it yet.

The "Export as CSV" link works in a similar way, it just sends the `GET` to an API call with appropriate parameters. The mime-type will cause the browser to download the CSV instead of trying to display it, so the navigation doesn't change from the admin UI.

# Checklist (Definition of Done)

- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/api/v1/admin/removals/route.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    if (searchParams.get("refresh") === "true") {
+      await refreshStuckRemovals(days);
+      return NextResponse.json({
+        success: true,
+        status: 200,
+        message: "refreshed stored results",
+      });
+    }
+
+    const { scanResults, brokers } = await getStuckRemovals(days, null, null);
```
Comment (mansaj) [inline src/app/api/v1/admin/removals/route.ts:38]: without try-catch blocks, we might swallow some errors / unhandled exceptions

File: src/app/api/v1/admin/removals/route.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        success: false,
+        status: 422,
+        message: "days parameter is required",
+      });
+    }
+
+    const days = parseInt(searchParams.get("days") ?? "30");
+
+    if (searchParams.get("refresh") === "true") {
+      await refreshStuckRemovals(days);
```
Comment (mansaj) [inline src/app/api/v1/admin/removals/route.ts:30]: without try-catch blocks, we might swallow some errors / unhandled exceptions

**Review by mansaj — CHANGES_REQUESTED** [review]

File: src/app/functions/server/getStuckRemovals.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { getAllDataBrokers } from "./onerep";
+import { refreshStoredScanResults } from "./refreshStoredScanResults";
+
+export async function getStuckRemovals(
+  days: number,
+  page: number | null,
+  perPage: number | null,
+) {
+  const age = new Date();
+  const dateOffset = 24 * 60 * 60 * 1000 * days;
```
Comment (flozia) [inline src/app/functions/server/getStuckRemovals.ts:15]: We have the const `CONST_DAY_MILLISECONDS`. :)

```suggestion
  const dateOffset = CONST_DAY_MILLISECONDS * days;
```

File: src/app/functions/server/getStuckRemovals.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      }) => [broker.id, broker.status],
+    ),
+  );
+
+  return { totalPages, scanResults, brokers };
+}
+
+export async function refreshStuckRemovals(days: number) {
+  const age = new Date();
+  const dateOffset = 24 * 60 * 60 * 1000 * days;
```
Comment (flozia) [inline src/app/functions/server/getStuckRemovals.ts:45]:
```suggestion
  const dateOffset = CONST_DAY_MILLISECONDS * days;
```

**Review by flozia — APPROVED** [review]
I was not able to test the `Pagination` component in detail. Considering this is only a page for internal testing only and we need to have it in place as soon as possible, this should be good to land.

**Review by mansaj — APPROVED** [review]
lgtm thanks

**Pull request merged**